lxd/instance/drivers/driver_qemu.go

```diff
@@ -1512,12 +1515,44 @@ func (vm *qemu) addRootDriveConfig(sb *strings.Builder, bootIndexes map[string]i
 		DevPath: rootDrivePath,
 	}
 
+	// If the storage pool is on ZFS and backed by a loop file then indicate to qemu we need to use asyncio.
+	// We use async I/O to avoid kernel hangs and qemu direct I/O errors with pre ZFS 0.8.
+	if pool.Driver().Info().Name == "zfs" && !shared.IsBlockdevPath(pool.Driver().Config()["source"]) {
```

Comment:
Source on my laptop is `castiana/lxd`, this path doesn't exist at all but should be handled as a block device, will that be the case?

Otherwise, we'd probably want something like PathExists combined with IsBlockDev

Reply:
@stgraber ah yes, because if its a relative import we dont want it to be treated as a loop back file. Will make the change.